tests/unit/type/stream.tcl

```diff
@@ -102,6 +102,30 @@ start_server {
         }
     }
 
+    test {XADD with MAXLEN option and the '=' argument} {
+        r DEL mystream
+        for {set j 0} {$j < 1000} {incr j} {
+            if {rand() < 0.9} {
+                r XADD mystream MAXLEN = 5 * xitem $j
+            } else {
+                r XADD mystream MAXLEN = 5 * yitem $j
+            }
+        }
+        assert {[r XLEN mystream] == 5}
+    }
+
+    test {XADD with MAXLEN option and the '~' argument} {
+        r DEL mystream
+        for {set j 0} {$j < 1000} {incr j} {
+            if {rand() < 0.9} {
+                r XADD mystream MAXLEN ~ 555 * xitem $j
+            } else {
+                r XADD mystream MAXLEN ~ 555 * yitem $j
+            }
+        }
+        assert {[r XLEN mystream] == 600}
```

Comment:
maybe we rather do `> 555 && < 1000` instead of `== 600`?
if we don't this test could some day break if something changes.
.. which is not so bad, we can then use that opportunity to modify it.

Reply:
I'm fine letting it break if/when this changes. Currently, the default is max_node 100 so that's why I used these numbers. Agreed?

Reply:
yeah